src/js/factual_background.js

```diff
@@ -234,7 +246,30 @@ class FactualBackground {
     }
 
     if (tabInfo.isContentLoaded()) {
+      // TODO to check: this might stop checking content on newly loaded Facebook posts
+      // PS. this check is not neccessary
       console.log('Content already loaded');
+
+      if (tabInfo.getCurrentUrl() !== tabInfo.getPreviousUrl()
+        && tabInfo.getCurrentUrl() && tabInfo.getPreviousUrl()) {
+
+        // URL changed, potential dynamic page reload, such as on YouTube
+        console.debug(`URL changed from ${tabInfo.getPreviousUrl()} to ${tabInfo.getCurrentUrl()}. Dynamic page reload?`);
+
+        var parsedCurrent = parseUrl(tabInfo.getCurrentUrl());
+        var parsedPrevious = parseUrl(tabInfo.getPreviousUrl());
+        parsedCurrent.hash = '';
+        parsedPrevious.hash = '';
+
+        if (parsedCurrent.href === parsedPrevious.href) {
+          // only hash changed
```

Comment:
Myślę, że ten warunek też trzeba będzie jakoś rozwiązać - wiele stron w technologii SPA korzysta ze zmian hasha przy nawigowaniu między podstronami więc mimo, że zminia się tylko hash może zmienić się też zawartość strony. Ale to już raczej nie bajka na ten PR ;)

Reply:
Poprawiłem to robiąc wyjątek na YouTube.